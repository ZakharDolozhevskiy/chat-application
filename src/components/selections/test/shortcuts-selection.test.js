import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { ShortcutsSelection } from '../shortcuts-selection';

describe('<ShortcutsSelection/>', () => {
  const callback = jest.fn();

  const component = shallow(
    <ShortcutsSelection
      className=""
      onChange={callback}
      translate={id => id}
      hotKeys={true}
    />
  );

  it('should render component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onChange callback if changes happen', () => {
    component.instance().onChange(null, true);
    expect(callback.mock.calls[0][0]).toBeTruthy();
  });
});