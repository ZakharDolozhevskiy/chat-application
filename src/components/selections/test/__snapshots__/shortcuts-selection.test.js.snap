// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortcutsSelection/> should render component 1`] = `
<div
  className=""
>
  <WithStyles(Typography)
    className="label"
    color="inherit"
    component="p"
  >
    Send message on CTRL+ENTER
    :
  </WithStyles(Typography)>
  <RadioGroup
    className="controls"
    onChange={[Function]}
    value="true"
  >
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Radio)
          className="selection-radio"
        />
      }
      label="On"
      value="true"
    />
    <WithStyles(FormControlLabel)
      control={
        <WithStyles(Radio)
          className="selection-radio"
        />
      }
      label="Off"
      value="false"
    />
  </RadioGroup>
</div>
`;
